gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/P.java

```diff
@@ -57,6 +57,11 @@ public V getOriginalValue() {
         return originalValue;
     }
 
+    /*
+     * Get the name of the predicate
+     */
+    public String getPredicateName() { return biPredicate.toString(); }
```

Comment:
just off the top of my head, `biPredicate` is usually an `Enum`. could we avoid future problems potentially if we tested for `Enum` and then called `name()` and restore to `toString()` for all other cases. I guess another question is whether a random `BiPredicate` even makes sense to have a predicate name? Like, is there a predicate name for a lambda and what use would that be in serialization? it would just cause an exception in serialization. is that the best way for that error to raise?